### PR TITLE
Prevent deleting or renaming the default profile

### LINKED ISSUE #4791: Prevent deleting or renaming the "default" profile
As we clearly assume the default profile is always a thing in a whole bunch of places, we should have the API reflect that and refuse any deletion or rename attempt for it.

### PR DESCRIPTION
**Pull request by monstermunchkin**
This resolves #4791

### CODE REVIEW

**Comment by stgraber** [comment]
@monstermunchkin I wouldn't do the API extension, this is a bugfix, deleting the profile today would cause a whole bunch of random issues, we're just making it so people don't shoot themselves in the foot.

File: test/suites/basic.sh [hunk 1]
```diff
@@ -477,4 +477,8 @@ test_basic_usage() {
 
   lxc stop foo --force || true
   ! lxc list | grep -q foo || false
+
+  # Test renaming/deletion of the default profile
+  lxc profile rename default foobar 2>&1 | grep -q "Error: The 'default' profile cannot be renamed"
+  lxc profile delete default 2>&1 | grep -q "Error: The 'default' profile cannot be deleted"
```
Comment (stgraber) [inline test/suites/basic.sh]: That should probably just be:
```
 ! lxc profile rename default foobar
 ! lxc profile delete default
```

File: test/includes/lxd.sh [hunk 2]
```diff
@@ -164,6 +164,23 @@ kill_lxd() {
...
         done
 
+        # Delete all devices of the default profile since the profile itself
+        # cannot be deleted.
+        echo "==> Deleting all devices of the default profile"
+        for device in $(lxc profile device list default --force-local); do
+            lxc profile device remove default "${device}" --force-local || true
+        done
+
+        echo "==> Deleting config of default profile"
```
Comment (stgraber) [inline test/includes/lxd.sh]: Can probably be done cleaner by doing `echo 'config: {}, devices: {}' | lxc profile edit default`

Comment (monstermunchkin) [reply test/includes/lxd.sh]: Fixed

**Comment by monstermunchkin** [comment]
I removed the API extension. Should be fine now.

**Comment by brauner** [comment]
@monstermunchkin, see my comments on irc.

**Pull request merged**